src/main/java/pl/piekoszek/gorskimatches/equation/SolutionToQuizzesMapper.java

```diff
@@ -25,19 +31,65 @@ public class SolutionToQuizzesMapper {
         numberOrSymbolChangeableToOther.put('+', new ArrayList<>());
         numberOrSymbolChangeableToOther.put('-', new ArrayList<>());
         numberOrSymbolChangeableToOther.put('=', new ArrayList<>());
-        numberOrSymbolChangeableToOther.put('4', new ArrayList<>());
         numberOrSymbolChangeableToOther.get('0').add('6');
         numberOrSymbolChangeableToOther.get('0').add('9');
         numberOrSymbolChangeableToOther.get('6').add('0');
         numberOrSymbolChangeableToOther.get('6').add('9');
         numberOrSymbolChangeableToOther.get('9').add('0');
         numberOrSymbolChangeableToOther.get('9').add('6');
-        numberOrSymbolChangeableToOther.get('3').add('5');
         numberOrSymbolChangeableToOther.get('3').add('2');
+        numberOrSymbolChangeableToOther.get('3').add('5');
         numberOrSymbolChangeableToOther.get('5').add('3');
         numberOrSymbolChangeableToOther.get('2').add('3');
     }
 
+    {
+        takeOneMatchFromNumberOrSymbol.put('0', new ArrayList<>());
+        takeOneMatchFromNumberOrSymbol.put('1', new ArrayList<>());
+        takeOneMatchFromNumberOrSymbol.put('2', new ArrayList<>());
+        takeOneMatchFromNumberOrSymbol.put('3', new ArrayList<>());
+        takeOneMatchFromNumberOrSymbol.put('4', new ArrayList<>());
+        takeOneMatchFromNumberOrSymbol.put('5', new ArrayList<>());
+        takeOneMatchFromNumberOrSymbol.put('6', new ArrayList<>());
+        takeOneMatchFromNumberOrSymbol.put('7', new ArrayList<>());
+        takeOneMatchFromNumberOrSymbol.put('8', new ArrayList<>());
```

Comment:
I'd suggest 
```
giveOneMatchFromNumberOrSymbol.put('5', List.of('6', '9'));
```
instead of 
```
giveOneMatchFromNumberOrSymbol.put('5', new ArrayList<>());
giveOneMatchFromNumberOrSymbol.get('5').add('6');
giveOneMatchFromNumberOrSymbol.get('5').add('9');
```
for all cases